<!DOCTYPE html>
<html lang="en"><head>
<script src="aula_5_rnaseq_24_files/libs/clipboard/clipboard.min.js"></script>
<script src="aula_5_rnaseq_24_files/libs/quarto-html/tabby.min.js"></script>
<script src="aula_5_rnaseq_24_files/libs/quarto-html/popper.min.js"></script>
<script src="aula_5_rnaseq_24_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="aula_5_rnaseq_24_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="aula_5_rnaseq_24_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="aula_5_rnaseq_24_files/libs/quarto-html/quarto-syntax-highlighting-29e2c20b02301cfff04dc8050bf30c7e.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.6.37">

  <meta name="author" content="James R. Hunter, PhD">
  <meta name="dcterms.date" content="2024-12-03">
  <title>Análise dos Dados com R</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="aula_5_rnaseq_24_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="aula_5_rnaseq_24_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="aula_5_rnaseq_24_files/libs/revealjs/dist/theme/quarto-f573e2b396c0ec73ba2f4e1283494cda.css">
  <link href="aula_5_rnaseq_24_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="aula_5_rnaseq_24_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="aula_5_rnaseq_24_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="aula_5_rnaseq_24_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Análise dos Dados com R</h1>
  <p class="subtitle">Bioconductor – RNASeq</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
James R. Hunter, PhD 
</div>
        <p class="quarto-title-affiliation">
            Retrovirologia, EPM, UNIFESP
          </p>
    </div>
</div>

  <p class="date">2024-12-03</p>
</section>
<section>
<section id="nóticia-importante" class="title-slide slide level1 center">
<h1>Nóticia Importante</h1>
<ul>
<li><p>Aula de 10 de dezembro (semana que vem) vai ser trocado para 8 de janeiro</p></li>
<li><p>Semana que vem Prof.&nbsp;Fernando Antonelli vai dar a 1ª aula dele</p></li>
</ul>
</section>
<section id="introdução" class="slide level2">
<h2>Introdução</h2>
<ul>
<li><p>Análise de dados da sequência de alto rendimento (“high-throughput sequence data”)</p></li>
<li><p>Investigar vários aspectos de genes e transcripts</p></li>
<li><p>Hoje, olhar na expressão diferencial dos genes</p></li>
<li><p>O processo vai definir um dos muitos <strong>fluxos de trabalho</strong> possíveis de análise bioinformática</p></li>
</ul>
</section>
<section id="pacotes-para-instalar-para-a-aula" class="slide level2">
<h2>Pacotes para Instalar para a Aula</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href=""></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">"airway"</span>, <span class="st">"tximeta"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Instalação desses 2 pacotes vai causar instalação de vários outros necessários para funcionamento</li>
</ul>
</section>
<section id="dados-airway" class="slide level2">
<h2>Dados – Airway</h2>
<ul>
<li><p>Um experimento RNA-seq que tratou células musculares lisas das vias aéreas com dexametasona</p></li>
<li><p>4 linhas de células</p>
<ul>
<li>Cada linha tem uma amostra tratada e outra não-tratada</li>
</ul></li>
<li><p>Dexametasona - glicocorticoide sintético esteroide com efeitos anti-inflamatórios</p></li>
<li><p>Utilizado no tratamento de doenças respiratórias entre outras coisas</p></li>
<li><p>Dados resumidos no pacote <code>airway</code></p></li>
</ul>
</section>
<section id="carregar-pacotes" class="slide level2">
<h2>Carregar Pacotes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href=""></a><span class="fu">library</span>(airway)</span>
<span id="cb2-2"><a href=""></a><span class="fu">library</span>(tximeta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="os-pacotes-carregados" class="slide level2 smaller">
<h2>Os Pacotes Carregados</h2>
<div class="cell">
<div class="cell-output-display">
<div id="gyjbtctcrn" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#gyjbtctcrn table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#gyjbtctcrn thead, #gyjbtctcrn tbody, #gyjbtctcrn tfoot, #gyjbtctcrn tr, #gyjbtctcrn td, #gyjbtctcrn th {
  border-style: none;
}

#gyjbtctcrn p {
  margin: 0;
  padding: 0;
}

#gyjbtctcrn .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gyjbtctcrn .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#gyjbtctcrn .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gyjbtctcrn .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gyjbtctcrn .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gyjbtctcrn .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gyjbtctcrn .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gyjbtctcrn .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gyjbtctcrn .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gyjbtctcrn .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gyjbtctcrn .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gyjbtctcrn .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gyjbtctcrn .gt_spanner_row {
  border-bottom-style: hidden;
}

#gyjbtctcrn .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#gyjbtctcrn .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gyjbtctcrn .gt_from_md > :first-child {
  margin-top: 0;
}

#gyjbtctcrn .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gyjbtctcrn .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gyjbtctcrn .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#gyjbtctcrn .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#gyjbtctcrn .gt_row_group_first td {
  border-top-width: 2px;
}

#gyjbtctcrn .gt_row_group_first th {
  border-top-width: 2px;
}

#gyjbtctcrn .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gyjbtctcrn .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#gyjbtctcrn .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#gyjbtctcrn .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gyjbtctcrn .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gyjbtctcrn .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gyjbtctcrn .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#gyjbtctcrn .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gyjbtctcrn .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gyjbtctcrn .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gyjbtctcrn .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gyjbtctcrn .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gyjbtctcrn .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gyjbtctcrn .gt_left {
  text-align: left;
}

#gyjbtctcrn .gt_center {
  text-align: center;
}

#gyjbtctcrn .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gyjbtctcrn .gt_font_normal {
  font-weight: normal;
}

#gyjbtctcrn .gt_font_bold {
  font-weight: bold;
}

#gyjbtctcrn .gt_font_italic {
  font-style: italic;
}

#gyjbtctcrn .gt_super {
  font-size: 65%;
}

#gyjbtctcrn .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#gyjbtctcrn .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#gyjbtctcrn .gt_indent_1 {
  text-indent: 5px;
}

#gyjbtctcrn .gt_indent_2 {
  text-indent: 10px;
}

#gyjbtctcrn .gt_indent_3 {
  text-indent: 15px;
}

#gyjbtctcrn .gt_indent_4 {
  text-indent: 20px;
}

#gyjbtctcrn .gt_indent_5 {
  text-indent: 25px;
}

#gyjbtctcrn .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#gyjbtctcrn div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_heading header">
<th colspan="3" class="gt_heading gt_title gt_font_normal gt_bottom_border">Tximeta &amp; Airway Related Packages</th>
</tr>
<tr class="gt_col_headings even">
<th id="package" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">package</th>
<th id="version" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">version</th>
<th id="date" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">date</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="package">airway</td>
<td class="gt_row gt_right" headers="version">1.26.0</td>
<td class="gt_row gt_right" headers="date">2024-10-31</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="package">Biobase</td>
<td class="gt_row gt_right" headers="version">2.66.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="package">BiocGenerics</td>
<td class="gt_row gt_right" headers="version">0.52.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="package">GenomeInfoDb</td>
<td class="gt_row gt_right" headers="version">1.42.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="package">GenomicRanges</td>
<td class="gt_row gt_right" headers="version">1.58.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="package">IRanges</td>
<td class="gt_row gt_right" headers="version">2.40.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="package">MatrixGenerics</td>
<td class="gt_row gt_right" headers="version">1.18.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="package">matrixStats</td>
<td class="gt_row gt_right" headers="version">1.4.1</td>
<td class="gt_row gt_right" headers="date">2024-09-08</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="package">S4Vectors</td>
<td class="gt_row gt_right" headers="version">0.44.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="package">SummarizedExperiment</td>
<td class="gt_row gt_right" headers="version">1.36.0</td>
<td class="gt_row gt_right" headers="date">2024-10-29</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="package">tximeta</td>
<td class="gt_row gt_right" headers="version">1.24.0</td>
<td class="gt_row gt_right" headers="date">2024-11-08</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<ul>
<li>Pedir 2, pega 11</li>
</ul>
</section>
<section id="passo-1-do-fluxo-quantificação" class="slide level2">
<h2>Passo 1 do Fluxo – Quantificação</h2>
<ul>
<li><p>Quantificação de expressão dos transcripts</p></li>
<li><p>Contagem feito fora de Bioconductor</p></li>
<li><p>Muitos softwares fazem isso</p></li>
<li><p>Este estudo utilizou <strong>Salmon</strong></p>
<ul>
<li><a href="https://combine-lab.github.io/salmon/" class="uri">https://combine-lab.github.io/salmon/</a></li>
</ul></li>
<li><p><code>airway</code> contem os arquivos saindo da operação de Salmon</p></li>
</ul>
</section>
<section id="arquivos-de-airway" class="slide level2 smaller">
<h2>Arquivos de <code>airway</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href=""></a>dir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">"extdata"</span>, <span class="at">package=</span><span class="st">"airway"</span>, <span class="at">mustWork=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-2"><a href=""></a><span class="fu">list.files</span>(dir) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "GSE52778_series_matrix.txt"        "Homo_sapiens.GRCh37.75_subset.gtf"
 [3] "quants"                            "sample_table.csv"                 
 [5] "SraRunInfo_SRP033351.csv"          "SRR1039508_subset.bam"            
 [7] "SRR1039509_subset.bam"             "SRR1039512_subset.bam"            
 [9] "SRR1039513_subset.bam"             "SRR1039516_subset.bam"            
[11] "SRR1039517_subset.bam"             "SRR1039520_subset.bam"            
[13] "SRR1039521_subset.bam"            </code></pre>
</div>
</div>
<ul>
<li><p>8 arquivos BAM</p></li>
<li><p>1 de referência</p></li>
<li><p>2 de contagem (na sub-pasta <code>quants</code>)</p></li>
</ul>
</section>
<section id="arquivos-com-contagem" class="slide level2">
<h2>Arquivos com Contagem</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href=""></a>(ff <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(dir, <span class="st">"quants"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "SRR1039508" "SRR1039509"</code></pre>
</div>
</div>
</section>
<section id="bam-file" class="slide level2">
<h2>BAM File</h2>
<ul>
<li><p><em>Binary Alignment Map</em></p></li>
<li><p>Os dados brutos abrangentes do sequênciamento genómico</p></li>
<li><p>Forma binária de um arquivo SAM (<em>Sequence Alignment Map</em>)</p></li>
</ul>
</section>
<section id="tabela-das-amostras" class="slide level2">
<h2>Tabela das Amostras</h2>
<ul>
<li><p>Próximo passo é criar um arquivo csv com Excel que liga as amostras para os arquivos associados FASTQ e Salmon.</p></li>
<li><p>Aqui já tem (<code>sample_table.csv</code>)</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href=""></a>csvfile <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="st">"sample_table.csv"</span>)</span>
<span id="cb7-2"><a href=""></a>coldata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(csvfile)</span>
<span id="cb7-3"><a href=""></a>coldata[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb7-4"><a href=""></a>coldata[, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">6</span><span class="sc">:</span><span class="dv">10</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section class="slide level2">

<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 5
  ...1       SampleName cell    dex   albut
  &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;
1 SRR1039508 GSM1275862 N61311  untrt untrt
2 SRR1039509 GSM1275863 N61311  trt   untrt
3 SRR1039512 GSM1275866 N052611 untrt untrt
4 SRR1039513 GSM1275867 N052611 trt   untrt
5 SRR1039516 GSM1275870 N080611 untrt untrt
6 SRR1039517 GSM1275871 N080611 trt   untrt
7 SRR1039520 GSM1275874 N061011 untrt untrt
8 SRR1039521 GSM1275875 N061011 trt   untrt</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 6
  ...1       Run        avgLength Experiment Sample    BioSample   
  &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;       
1 SRR1039508 SRR1039508       126 SRX384345  SRS508568 SAMN02422669
2 SRR1039509 SRR1039509       126 SRX384346  SRS508567 SAMN02422675
3 SRR1039512 SRR1039512       126 SRX384349  SRS508571 SAMN02422678
4 SRR1039513 SRR1039513        87 SRX384350  SRS508572 SAMN02422670
5 SRR1039516 SRR1039516       120 SRX384353  SRS508575 SAMN02422682
6 SRR1039517 SRR1039517       126 SRX384354  SRS508576 SAMN02422673
7 SRR1039520 SRR1039520       101 SRX384357  SRS508579 SAMN02422683
8 SRR1039521 SRR1039521        98 SRX384358  SRS508580 SAMN02422677</code></pre>
</div>
</div>
</section>
<section id="trabalhar-com-salmon-quant-data" class="slide level2">
<h2>Trabalhar com Salmon Quant Data</h2>
<ul>
<li><p>Só as primeiras duas amostras são os dados de Salmon.</p>
<ul>
<li>Vamos trabalhar com eles</li>
</ul></li>
<li><p>Dá colunas nomes de <code>names</code> e <code>files</code></p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href=""></a>coldata12 <span class="ot">&lt;-</span> coldata[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,]</span>
<span id="cb10-2"><a href=""></a>coldata12<span class="sc">$</span>names <span class="ot">&lt;-</span> coldata12<span class="sc">$</span>Run</span>
<span id="cb10-3"><a href=""></a>coldata12<span class="sc">$</span>files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="st">"quants"</span>, coldata12<span class="sc">$</span>names, <span class="st">"quant.sf.gz"</span>)</span>
<span id="cb10-4"><a href=""></a><span class="fu">file.exists</span>(coldata12<span class="sc">$</span>files)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href=""></a>coldata12<span class="sc">$</span>files</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/Users/jameshunter/Library/R/arm64/4.4/library/airway/extdata/quants/SRR1039508/quant.sf.gz"
[2] "/Users/jameshunter/Library/R/arm64/4.4/library/airway/extdata/quants/SRR1039509/quant.sf.gz"</code></pre>
</div>
</div>
</section>
<section id="importar-os-dados-para-preparar-eles" class="slide level2 smaller">
<h2>Importar os Dados para Preparar Eles</h2>
<ul>
<li><p><code>tximeta</code> importa metadata e anotações durante a importação das quantificações dos transcripts</p></li>
<li><p>Importa dados ao nível dos transcripts</p></li>
<li><p>tximeta() localiza e faz o download das informações dos fontes vários do internet.</p></li>
<li><p>O resultado do trabalho é um objeto da classe <code>SummarizedExperiment</code> (<code>se</code>)</p></li>
</ul>
<blockquote>
<p>Tximeta performs numerous annotation and metadata gathering tasks on behalf of users during the import of transcript quantifications from <em>Salmon</em>, <em>alevin</em>, or <em>piscem-infer</em> into R/Bioconductor. Metadata and transcript ranges are added automatically, facilitating genomic analyses and assisting in computational reproducibility.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href=""></a>se <span class="ot">&lt;-</span> <span class="fu">tximeta</span>(coldata12)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="tximeta-retorna-a-seguinte-informação" class="slide level2 smaller">
<h2>tximeta Retorna a Seguinte Informação</h2>

<img data-src="../images/tximeta_output.png" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="tamanho-do-conjunto-dos-dados" class="slide level2">
<h2>Tamanho do Conjunto dos Dados</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href=""></a><span class="fu">dim</span>(se)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 205870      2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href=""></a><span class="fu">head</span>(<span class="fu">rownames</span>(se)) <span class="co"># 1a 6 fileiras</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENST00000456328.2" "ENST00000450305.2" "ENST00000488147.1"
[4] "ENST00000619216.1" "ENST00000473358.1" "ENST00000469289.1"</code></pre>
</div>
</div>
<ul>
<li><p>Total de 110.4 MB na memoria</p></li>
<li><p>205870 Transcripts</p></li>
<li><p>Precisa fazer um resumo das quantificações (que são ao nível do transcript) ao nível dos genes.</p>
<ul>
<li>Objetivo: Analisar os genes</li>
</ul></li>
<li><p>Função <code>summarizetoGene()</code> faz a conversão para o nível do gene</p></li>
</ul>
</section></section>
<section>
<section id="summarizedexperiment" class="title-slide slide level1 center" data-background-color="blue">
<h1>SummarizedExperiment</h1>
<ul>
<li>Classe geral que facilita a transferência dos dados entre pacotes</li>
</ul>
</section>
<section id="summarizedexperiment-3-blocos-de-informação" class="slide level2">
<h2><code>SummarizedExperiment</code> – 3 Blocos de Informação</h2>

<img data-src="../images/se_blocks.png" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="blocos" class="slide level2">
<h2>3 Blocos</h2>
<ul>
<li><p>rowRanges: info re: os intervalos genômicos</p></li>
<li><p>colData: info re: as amostras</p></li>
<li><p><em>“assays”/counts</em> matriz da contagem</p>
<ul>
<li>Os dados que vamos usar para a análise</li>
</ul></li>
</ul>
</section>
<section id="converter-ao-um-summarizedexperiment-dos-genes" class="slide level2 smaller">
<h2>Converter ao Um <code>SummarizedExperiment</code> dos Genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href=""></a>gse <span class="ot">&lt;-</span> <span class="fu">summarizeToGene</span>(se)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="../images/gse_result.png" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="verificar-tamanho-dos-genes" class="slide level2">
<h2>Verificar Tamanho dos Genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href=""></a><span class="fu">dim</span>(gse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 58294     2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href=""></a><span class="fu">head</span>(<span class="fu">rownames</span>(gse))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENSG00000000003.14" "ENSG00000000005.5"  "ENSG00000000419.12"
[4] "ENSG00000000457.13" "ENSG00000000460.16" "ENSG00000000938.12"</code></pre>
</div>
</div>
<ul>
<li>58,294 genes invés de 205,870 transcripts e só 53 MB</li>
</ul>
</section></section>
<section>
<section id="pronto-para-deseq2" class="title-slide slide level1 center" data-background-color="blue">
<h1>Pronto para dESeq2</h1>
<ul>
<li><p>Verdadeira análise</p></li>
<li><p>O fluxo de trabalho que usamos é só um. Existem vários</p></li>
</ul>
</section>
<section id="versão-final-de-gse" class="slide level2">
<h2>Versão Final de <code>gse</code></h2>
<ul>
<li>Vamos usar um outra versão de <code>gse</code> que vem do pacote (agora com 38MB)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href=""></a><span class="fu">data</span>(gse) <span class="co">#por na memoria do pacote</span></span>
<span id="cb24-2"><a href=""></a>gse</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RangedSummarizedExperiment 
dim: 58294 8 
metadata(6): tximetaInfo quantInfo ... txomeInfo txdbInfo
assays(3): counts abundance length
rownames(58294): ENSG00000000003.14 ENSG00000000005.5 ...
  ENSG00000285993.1 ENSG00000285994.1
rowData names(1): gene_id
colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
colData names(3): names donor condition</code></pre>
</div>
</div>
</section>
<section id="olhar-na-matriz-counts" class="slide level2">
<h2>Olhar na Matriz <code>counts</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href=""></a><span class="fu">assayNames</span>(gse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "counts"    "abundance" "length"   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href=""></a><span class="fu">head</span>(<span class="fu">assay</span>(gse), <span class="dv">3</span>) <span class="co"># só 3 fileiras</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
ENSG00000000003.14    708.164    467.962    900.992    424.368   1188.295
ENSG00000000005.5       0.000      0.000      0.000      0.000      0.000
ENSG00000000419.12    455.000    510.000    604.000    352.000    583.000
                   SRR1039517 SRR1039520 SRR1039521
ENSG00000000003.14   1090.668    805.929    599.337
ENSG00000000005.5       0.000      0.000      0.000
ENSG00000000419.12    773.999    409.999    499.000</code></pre>
</div>
</div>
<ul>
<li>Os elementos dos dados que são o objetivo do estudo</li>
</ul>
</section>
<section id="totais-das-colunas" class="slide level2">
<h2>Totais das Colunas</h2>
<ul>
<li><p>Através de todas as amostras</p></li>
<li><p>Nova função <code>colSums()</code></p>
<ul>
<li>Faz total de todos os elementos de todos os colunas</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href=""></a><span class="fu">colSums</span>(<span class="fu">assay</span>(gse))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 
  21100805   19298584   26145537   15688246   25268618   31891456   19683767 
SRR1039521 
  21813903 </code></pre>
</div>
</div>
</section>
<section id="rowranges" class="slide level2">
<h2>rowRanges</h2>
<ul>
<li><p>Informação sobre os genes (fileiras)</p></li>
<li><p>No formato de GRanges</p>
<ul>
<li><p>Classe para organizar genes ou transcripts mostrando os dados</p>
<ul>
<li><p>Extenso do gene (Start - Stop)</p></li>
<li><p>Metadata</p></li>
<li><p>Sequências (cromossomos)</p></li>
</ul></li>
</ul></li>
<li><p>Comando mostra os dados sobre os primeiro 5 e último 5 ranges</p></li>
</ul>
</section>
<section class="slide level2">

<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href=""></a><span class="fu">rowRanges</span>(gse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 58294 ranges and 1 metadata column:
                     seqnames              ranges strand |            gene_id
                        &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |        &lt;character&gt;
  ENSG00000000003.14     chrX 100627109-100639991      - | ENSG00000000003.14
   ENSG00000000005.5     chrX 100584802-100599885      + |  ENSG00000000005.5
  ENSG00000000419.12    chr20   50934867-50958555      - | ENSG00000000419.12
  ENSG00000000457.13     chr1 169849631-169894267      - | ENSG00000000457.13
  ENSG00000000460.16     chr1 169662007-169854080      + | ENSG00000000460.16
                 ...      ...                 ...    ... .                ...
   ENSG00000285990.1    chr14   19244904-19269380      - |  ENSG00000285990.1
   ENSG00000285991.1     chr6 149817937-149896011      - |  ENSG00000285991.1
   ENSG00000285992.1     chr8   47129262-47132628      + |  ENSG00000285992.1
   ENSG00000285993.1    chr18   46409197-46410645      - |  ENSG00000285993.1
   ENSG00000285994.1    chr10   12563151-12567351      + |  ENSG00000285994.1
  -------
  seqinfo: 25 sequences (1 circular) from hg38 genome</code></pre>
</div>
</div>
</section>
<section id="informação-sobre-as-sequências" class="slide level2 smaller">
<h2>Informação sobre as Sequências</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href=""></a><span class="fu">seqinfo</span>(<span class="fu">rowRanges</span>(gse))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Seqinfo object with 25 sequences (1 circular) from hg38 genome:
  seqnames seqlengths isCircular genome
  chr1      248956422      FALSE   hg38
  chr2      242193529      FALSE   hg38
  chr3      198295559      FALSE   hg38
  chr4      190214555      FALSE   hg38
  chr5      181538259      FALSE   hg38
  ...             ...        ...    ...
  chr21      46709983      FALSE   hg38
  chr22      50818468      FALSE   hg38
  chrX      156040895      FALSE   hg38
  chrY       57227415      FALSE   hg38
  chrM          16569       TRUE   hg38</code></pre>
</div>
</div>
</section>
<section id="coldata-info-re-amostras" class="slide level2">
<h2>colData – Info re: Amostras</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href=""></a><span class="fu">colData</span>(gse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 8 rows and 3 columns
                names    donor     condition
             &lt;factor&gt; &lt;factor&gt;      &lt;factor&gt;
SRR1039508 SRR1039508  N61311  Untreated    
SRR1039509 SRR1039509  N61311  Dexamethasone
SRR1039512 SRR1039512  N052611 Untreated    
SRR1039513 SRR1039513  N052611 Dexamethasone
SRR1039516 SRR1039516  N080611 Untreated    
SRR1039517 SRR1039517  N080611 Dexamethasone
SRR1039520 SRR1039520  N061011 Untreated    
SRR1039521 SRR1039521  N061011 Dexamethasone</code></pre>
</div>
</div>
</section></section>
<section>
<section id="deseq2-nossa-ferramenta" class="title-slide slide level1 center">
<h1>DESeq2 – Nossa Ferramenta</h1>
<ul>
<li>Entre muitas outras</li>
</ul>
</section>
<section id="cria-uma-classe-deseqdataset" class="slide level2">
<h2>Cria uma classe <code>DESeqDataSet</code></h2>
<ul>
<li><p>Construído acima de um <code>SummarizedExperiment</code></p></li>
<li><p>Agora, o slot <code>assay</code> será acessado por <code>counts</code></p></li>
<li><p>Tem uma formula de design associado</p></li>
<li><p>Podemos acessar os dados das colunas com <code>$</code></p>
<ul>
<li><p>Aplica igualmente a <code>SummarizedExperiment</code> e <code>DESeqDataSet</code></p></li>
<li><p>Mesmo anotação que os data frames</p></li>
</ul></li>
</ul>
</section>
<section id="pode-ver-a-hierarquia-dos-slots-com-view" class="slide level2">
<h2>Pode Ver a Hierarquia dos Slots com <code>View()</code></h2>

<img data-src="../images/gse_str_view.png" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="algumas-colunas-de-gse" class="slide level2">
<h2>Algumas Colunas de <code>gse</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href=""></a>gse<span class="sc">$</span>donor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] N61311  N61311  N052611 N052611 N080611 N080611 N061011 N061011
Levels: N052611 N061011 N080611 N61311</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href=""></a>gse<span class="sc">$</span>condition</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Untreated     Dexamethasone Untreated     Dexamethasone Untreated    
[6] Dexamethasone Untreated     Dexamethasone
Levels: Untreated Dexamethasone</code></pre>
</div>
</div>
<blockquote>
<p>Qual tipo de dado são <code>donor</code> e <code>condition</code>?</p>
</blockquote>
</section>
<section id="pode-manipular-as-variáveis" class="slide level2 smaller">
<h2>Pode Manipular as Variáveis</h2>
<ul>
<li><p><code>donor</code> não faz muito sentido em nosso contexto: trocar para <code>cell</code></p></li>
<li><p>Trocar <code>dex</code> no lugar de <code>condition</code></p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href=""></a>gse<span class="sc">$</span>cell <span class="ot">&lt;-</span> gse<span class="sc">$</span>donor</span>
<span id="cb42-2"><a href=""></a>gse<span class="sc">$</span>dex <span class="ot">&lt;-</span> gse<span class="sc">$</span>condition</span>
<span id="cb42-3"><a href=""></a></span>
<span id="cb42-4"><a href=""></a><span class="co"># resultado</span></span>
<span id="cb42-5"><a href=""></a>gse<span class="sc">@</span>colData<span class="sc">@</span>listData</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$names
[1] SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520
[8] SRR1039521
8 Levels: SRR1039508 SRR1039509 SRR1039512 SRR1039513 ... SRR1039521

$donor
[1] N61311  N61311  N052611 N052611 N080611 N080611 N061011 N061011
Levels: N052611 N061011 N080611 N61311

$condition
[1] Untreated     Dexamethasone Untreated     Dexamethasone Untreated    
[6] Dexamethasone Untreated     Dexamethasone
Levels: Untreated Dexamethasone

$cell
[1] N61311  N61311  N052611 N052611 N080611 N080611 N061011 N061011
Levels: N052611 N061011 N080611 N61311

$dex
[1] Untreated     Dexamethasone Untreated     Dexamethasone Untreated    
[6] Dexamethasone Untreated     Dexamethasone
Levels: Untreated Dexamethasone</code></pre>
</div>
</div>
</section>
<section id="podemos-tirar-as-2-variáveis-que-não-prestam-mais" class="slide level2 smaller">
<h2>Podemos Tirar as 2 Variáveis que não Prestam Mais</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href=""></a>gse<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb44-2"><a href=""></a>gse<span class="sc">$</span>donor <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb44-3"><a href=""></a></span>
<span id="cb44-4"><a href=""></a><span class="co"># resultado</span></span>
<span id="cb44-5"><a href=""></a>gse<span class="sc">@</span>colData<span class="sc">@</span>listData</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$names
[1] SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520
[8] SRR1039521
8 Levels: SRR1039508 SRR1039509 SRR1039512 SRR1039513 ... SRR1039521

$cell
[1] N61311  N61311  N052611 N052611 N080611 N080611 N061011 N061011
Levels: N052611 N061011 N080611 N61311

$dex
[1] Untreated     Dexamethasone Untreated     Dexamethasone Untreated    
[6] Dexamethasone Untreated     Dexamethasone
Levels: Untreated Dexamethasone</code></pre>
</div>
</div>
</section>
<section id="pode-manipular-os-níveis" class="slide level2">
<h2>Pode Manipular os Níveis</h2>
<ul>
<li><p>Função (base R) levels() relata os níveis dos fatores</p>
<ul>
<li>Tb, pode receber novos valores</li>
</ul></li>
<li><p>VSS - a ordem dos níveis devem ser preservada</p></li>
<li><p>Para <code>dex</code></p>
<ul>
<li><p>Trocar “Untreated” para “untrt”</p></li>
<li><p>Trocar “Dexamethasone” para “trt”</p></li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href=""></a><span class="fu">levels</span>(gse<span class="sc">$</span>dex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Untreated"     "Dexamethasone"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href=""></a><span class="fu">levels</span>(gse<span class="sc">$</span>dex) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"untrt"</span>, <span class="st">"trt"</span>)</span>
<span id="cb48-2"><a href=""></a><span class="fu">levels</span>(gse<span class="sc">$</span>dex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "untrt" "trt"  </code></pre>
</div>
</div>
</section>
<section id="design-formula" class="slide level2 smaller">
<h2>Design Formula</h2>
<ul>
<li><p>Este elemento usamos para contar para DESeq2 o que queremos testar</p></li>
<li><p>É o tipo de anotação que usamos em muitas funções</p>
<ul>
<li>E.g., <code>lm()</code>, a função que usamos para fazer regressão linear</li>
</ul></li>
<li><p>Queremos saber para quais genes o efeito de tratamento é diferente</p></li>
<li><p>Normalmente tem formato de: <code>~ group + treatment</code></p>
<ul>
<li><p>Pode incluir um termo de interação, se for desejado</p></li>
<li><p><code>~ group + treatment + group:treatment</code></p></li>
<li><p>Não em nosso caso</p></li>
</ul></li>
</ul>
</section>
<section id="lembre-o-tamanho-dos-dados" class="slide level2">
<h2>Lembre o Tamanho dos Dados</h2>
<ul>
<li><p>Salmon mapeou milhões de fragmentos genômicos</p>
<ul>
<li>Abaixo, expresso em milhões de fragmentos</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href=""></a><span class="fu">round</span>( <span class="fu">colSums</span>(<span class="fu">assay</span>(gse)) <span class="sc">/</span> <span class="fl">1e6</span>, <span class="dv">1</span> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 
      21.1       19.3       26.1       15.7       25.3       31.9       19.7 
SRR1039521 
      21.8 </code></pre>
</div>
</div>
</section>
<section id="carregar-deseq2" class="slide level2">
<h2>Carregar DESeq2</h2>
<ul>
<li><p>Chamar a função <code>DESeqDataSet()</code> para preparar as operações de algoritmo</p></li>
<li><p>Usando o design formula</p></li>
<li><p>Produz um “<code>Large DESeqDataSet (58294 elements, 36.1 MB)</code>”</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href=""></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb52-2"><a href=""></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(gse, <span class="at">design =</span> <span class="sc">~</span> cell <span class="sc">+</span> dex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section>
<section>
<section id="exploração-de-dds" class="title-slide slide level1 center" data-background-color="blue">
<h1>Exploração de <code>dds</code></h1>
<ul>
<li><p>Antes de fazer a análise vamos explorar um pouco nossos dados</p></li>
<li><p>Especialmente as contagens</p></li>
</ul>
</section>
<section id="só-queremos-genes-que-dá-informação" class="slide level2">
<h2>Só Queremos Genes que Dá Informação</h2>
<ul>
<li><p>Filtrar as fileiras dos <code>counts</code> que têm suficiente fragmentos para informar nossa análise</p>
<ul>
<li>Muitas genes vão ter um número tão pequeno de fragmentos que são inúteis para a análise</li>
</ul></li>
<li><p>Vamos olhar um pouco na estrutura de counts</p></li>
</ul>
</section>
<section id="estrutura-de-counts" class="slide level2 smaller">
<h2>Estrutura de <code>counts</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href=""></a><span class="fu">str</span>(<span class="fu">counts</span>(dds))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> int [1:58294, 1:8] 708 0 455 312 88 0 3228 2298 549 427 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:58294] "ENSG00000000003.14" "ENSG00000000005.5" "ENSG00000000419.12" "ENSG00000000457.13" ...
  ..$ : chr [1:8] "SRR1039508" "SRR1039509" "SRR1039512" "SRR1039513" ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href=""></a><span class="co"># olhar no head de counts</span></span>
<span id="cb55-2"><a href=""></a><span class="fu">counts</span>(dds)[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
ENSG00000000003.14        708        468        901        424       1188
ENSG00000000005.5           0          0          0          0          0
ENSG00000000419.12        455        510        604        352        583
ENSG00000000457.13        312        268        363        224        361
ENSG00000000460.16         88         74         52         45        107
ENSG00000000938.12          0          0          2          0          1
                   SRR1039517 SRR1039520 SRR1039521
ENSG00000000003.14       1091        806        599
ENSG00000000005.5           0          0          0
ENSG00000000419.12        774        410        499
ENSG00000000457.13        430        300        289
ENSG00000000460.16        101         97         83
ENSG00000000938.12          0          0          0</code></pre>
</div>
</div>
</section>
<section id="proposta" class="slide level2 smaller">
<h2>Proposta</h2>
<ul>
<li><p>Pelo menos, 4 células de cada gene (fileira) deve ter ao menos 10 fragmentos</p>
<ul>
<li><p>fragmentos = count da célula</p></li>
<li><p>4 células porque temos 4 grupos que estamos avaliando</p>
<ul>
<li>4 linhas de células diferentes</li>
</ul></li>
</ul></li>
<li><p><code>rowSums()</code> calcula o total dos valores dentro de uma fileira, para todas as fileiras.</p></li>
<li><p>Queremos ter então pelo menos metade das colunas (4 de 8) com uma contagem &gt;= 10</p></li>
<li><p>Vai fazer o processamento mais rápido e reduzir o tamanho do DESeqDataSet</p></li>
</ul>
</section>
<section id="para-chegar-lá-.-.-." class="slide level2">
<h2>Para Chegar Lá . . .</h2>
<ul>
<li><p>Quantas fileiras têm uma contagem de 0</p>
<ul>
<li>Nenhuma amostra teve fragmento sobreposto do gene</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href=""></a>zeros <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">==</span> <span class="dv">0</span></span>
<span id="cb57-2"><a href=""></a>ddsz <span class="ot">&lt;-</span> dds[zeros, ]</span>
<span id="cb57-3"><a href=""></a><span class="fu">dim</span>(ddsz)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 22721     8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href=""></a>moreten <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb59-2"><a href=""></a>ddsmt <span class="ot">&lt;-</span> dds[moreten, ]</span>
<span id="cb59-3"><a href=""></a><span class="fu">dim</span>(ddsmt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 24260     8</code></pre>
</div>
</div>
</section>
<section id="para-completar-o-cálculo" class="slide level2 smaller">
<h2>Para Completar o Cálculo</h2>
<ul>
<li>Vamos ver quantos de <code>ddsmt</code> tem pelo menos 4 amostras com esta contagem</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href=""></a>keep <span class="ot">&lt;-</span> (<span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>)) <span class="sc">&gt;=</span> <span class="dv">4</span></span>
<span id="cb61-2"><a href=""></a><span class="co"># parênteses em torno do 1º teste para facilitar a leitura</span></span>
<span id="cb61-3"><a href=""></a>dds_fin <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb61-4"><a href=""></a><span class="co"># tamanho</span></span>
<span id="cb61-5"><a href=""></a><span class="fu">dim</span>(dds_fin)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 16637     8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href=""></a><span class="co"># mostrar cabeça de novo</span></span>
<span id="cb63-2"><a href=""></a><span class="fu">counts</span>(dds_fin)[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
ENSG00000000003.14        708        468        901        424       1188
ENSG00000000419.12        455        510        604        352        583
ENSG00000000457.13        312        268        363        224        361
ENSG00000000460.16         88         74         52         45        107
ENSG00000000971.15       3228       3655       6066       4210       6636
ENSG00000001036.13       2298       1630       2682       1357       2209
                   SRR1039517 SRR1039520 SRR1039521
ENSG00000000003.14       1091        806        599
ENSG00000000419.12        774        410        499
ENSG00000000457.13        430        300        289
ENSG00000000460.16        101         97         83
ENSG00000000971.15      10855       5032       7771
ENSG00000001036.13       2214       2115       1736</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href=""></a><span class="fu">dim</span>(dds_fin)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 16637     8</code></pre>
</div>
</div>
</section>
<section id="conclusões-deste-passo" class="slide level2">
<h2>Conclusões deste Passo</h2>
<ol type="1">
<li>Agora chegamos com 16.637 genes que têm informação útil</li>
</ol>
<!-- -->
<ol start="2" type="1">
<li>Velho ENSG00000000005.5 não é mais na 2ª fileira – contagem 0</li>
</ol>
</section>
<section id="fluxo-para-expressão-diferencial" class="slide level2">
<h2>Fluxo para Expressão Diferencial</h2>
<ul>
<li><p>Especificamos um desenho experimental na <em>design formula</em></p>
<ul>
<li>Quando criamos o <code>DESeqDataSet</code> : “<code>DESeqDataSet(gse, design = ~ cell + dex)</code>”</li>
</ul></li>
<li><p>Utilizando as contagens brutas que já temos, podemos chamar a função <code>DESeq</code></p></li>
<li><p>Retorna um novo DESeqDataSet com todos os parâmetros da análise</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href=""></a>dds_out <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_fin)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="conteúdo-dos-resultados-dds_fin" class="slide level2 smaller">
<h2>Conteúdo dos Resultados – <code>dds_fin</code></h2>
<ul>
<li><p>Objetivo da função: calcular a expressão diferencial de tratamento ou não-tratamento dos genes</p></li>
<li><p>Produz um serie de resultados num data frame</p></li>
<li><p>A expressão diferencial está diretamente visto na coluna “<code>log2FoldChange</code>” que está uma transformação das contagens</p>
<ul>
<li>Controla para diferenças em na profundidade do sequenciamento das amostras e a disperção do cada gene</li>
</ul></li>
<li><p>Aplica um modelo linear geral com a distribuição “<em>Negative Binomial</em>”</p></li>
<li><p>Indicações estatísticas sobre a diferencial para cada gene</p></li>
</ul>
</section>
<section id="tabela-dos-resultados" class="slide level2 smaller">
<h2>Tabela dos Resultados</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href=""></a>(res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_out))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): dex trt vs untrt 
Wald test p-value: dex trt vs untrt 
DataFrame with 16637 rows and 6 columns
                    baseMean log2FoldChange     lfcSE      stat         pvalue
                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;      &lt;numeric&gt;
ENSG00000000003.14  740.1093      -0.365327 0.1073385 -3.403501 0.000665281929
ENSG00000000419.12  511.6990       0.202232 0.1278579  1.581694 0.113719502671
ENSG00000000457.13  314.1680       0.033792 0.1552106  0.217717 0.827649766335
ENSG00000000460.16   79.7988      -0.120633 0.3055270 -0.394836 0.692964158474
ENSG00000000971.15 5715.3064       0.442982 0.0904089  4.899766 0.000000959508
...                      ...            ...       ...       ...            ...
ENSG00000285953.1    29.5747      -1.920562  0.649661 -2.956253     0.00311401
ENSG00000285967.1   181.1650      -0.325885  0.179340 -1.817132     0.06919688
ENSG00000285976.1   875.4424       0.262132  0.142980  1.833351     0.06675037
ENSG00000285979.1    38.3502       0.338383  0.348445  0.971124     0.33148631
ENSG00000285991.1    11.2772      -0.115472  0.723139 -0.159681     0.87313221
                           padj
                      &lt;numeric&gt;
ENSG00000000003.14 0.0046355222
ENSG00000000419.12 0.2881253592
ENSG00000000457.13 0.9217179061
ENSG00000000460.16 0.8487430958
ENSG00000000971.15 0.0000136387
...                         ...
ENSG00000285953.1     0.0171889
ENSG00000285967.1     0.2019448
ENSG00000285976.1     0.1965808
ENSG00000285979.1     0.5713913
ENSG00000285991.1            NA</code></pre>
</div>
</div>
<ul>
<li>Quer saber o querem dizer essas colunas?</li>
</ul>
</section>
<section id="metadata-na-res" class="slide level2">
<h2>Metadata na <code>res</code></h2>
<ul>
<li><p>Função do pacote <code>S4Vectors</code></p></li>
<li><p>Especificamente acessar as colunas com metadata</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href=""></a><span class="fu">mcols</span>(res, <span class="at">use.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 6 rows and 2 columns
                       type            description
                &lt;character&gt;            &lt;character&gt;
baseMean       intermediate mean of normalized c..
log2FoldChange      results log2 fold change (ML..
lfcSE               results standard error: dex ..
stat                results Wald statistic: dex ..
pvalue              results Wald test p-value: d..
padj                results   BH adjusted p-values</code></pre>
</div>
</div>
</section>
<section id="log2foldchange" class="slide level2">
<h2><code>log2FoldChange</code></h2>
<ul>
<li><p>Estimativa de tamanho de efeito (<em>effect size</em>)</p></li>
<li><p>Relata quanto a expressão do gene mudou por causa do tratamento com dexametasona</p></li>
<li><p>Escala logaritmica relativa a base 2</p></li>
</ul>
<p><span class="math display">\[log2FoldChange = 1.5 = 2^{1.5}\approx2.828 \]</span></p>
<ul>
<li><p><code>lfcSE</code> – erro padrão de <em>fold change</em></p></li>
<li><p>Outros – estatísticas</p></li>
</ul>
</section>
<section id="resumo-dos-resultados" class="slide level2 smaller">
<h2>Resumo dos Resultados</h2>
<ul>
<li><p>A função generica <code>summary()</code> tem método especial aqui para classe <code>DESeqResults</code></p></li>
<li><p>Pode ajustar o valor-p com o argumento <code>alpha =</code> e um valor (e.g.&nbsp;0.05)</p></li>
<li><p>Temos aqui 4381 genes mostrando diferencial dos 16637 genes</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href=""></a><span class="fu">summary</span>(res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 16637 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 2362, 14%
LFC &lt; 0 (down)     : 2019, 12%
outliers [1]       : 0, 0%
low counts [2]     : 646, 3.9%
(mean count &lt; 12)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
</div>
</section>
<section id="mudança-para-alpha-de-0.05" class="slide level2">
<h2>Mudança para <span class="math inline">\(\alpha\)</span> de 0.05</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href=""></a>res05 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_out, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb74-2"><a href=""></a><span class="fu">table</span>(res05<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
12712  3602 </code></pre>
</div>
</div>
<ul>
<li>Redução do número de genes que mostram a diferencial significativa</li>
</ul>
</section></section>
<section>
<section id="gráficos-dos-resultados" class="title-slide slide level1 center" data-background-color="blue">
<h1>Gráficos dos Resultados</h1>
<ul>
<li>Uma amostra das possibilidades gráficas</li>
</ul>
</section>
<section id="gráfico-das-contagens-brutas" class="slide level2">
<h2>Gráfico das Contagens Brutas</h2>
<ul>
<li><p>Queremos saber com o gene mais significativa mudou por causa de tratamento</p></li>
<li><p>Vamos indentificar o gene mais significativa, i.e., o gene com o valor mínimo de <code>padj</code></p></li>
<li><p>Mostrar as contagens brutas relacionados a ele</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href=""></a>(topgene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)[<span class="fu">which.min</span>(res<span class="sc">$</span>padj)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENSG00000189221.9"</code></pre>
</div>
</div>
</section>
<section id="construir-gráfico" class="slide level2 smaller">
<h2>Construir Gráfico</h2>
<ul>
<li><p>Precisa <code>DESeq2::plotCounts()</code> – função que prepara contagens para gráficos fazendo</p>
<ul>
<li><p>Normalização</p></li>
<li><p>Mais um <em>pseudocontagem</em> de 0.5 para evitar logs de 0</p></li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href=""></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb78-2"><a href=""></a>geneCounts <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds_out, <span class="at">gene =</span> topgene, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"dex"</span>, <span class="st">"cell"</span>),</span>
<span id="cb78-3"><a href=""></a>                         <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span>
<span id="cb78-4"><a href=""></a><span class="fu">head</span>(geneCounts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               count   dex    cell
SRR1039508  443.7641 untrt  N61311
SRR1039509 4570.3854   trt  N61311
SRR1039512  470.3890 untrt N052611
SRR1039513 4403.9701   trt N052611
SRR1039516  518.8819 untrt N080611
SRR1039517 4594.8870   trt N080611</code></pre>
</div>
</div>
</section>
<section id="o-gráfico" class="slide level2">
<h2>O Gráfico</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href=""></a><span class="fu">ggplot</span>(geneCounts, <span class="fu">aes</span>(<span class="at">x =</span> dex, <span class="at">y =</span> count, <span class="at">color =</span> cell)) <span class="sc">+</span></span>
<span id="cb80-2"><a href=""></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb80-3"><a href=""></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb80-4"><a href=""></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="aula_5_rnaseq_24_files/figure-revealjs/top_graf-1.png" style="width:100.0%" class="r-stretch"></section>
<section class="slide level2">

<h3 id="para-fazer-mais-claro-acrescentar-linhas">Para Fazer Mais Claro, Acrescentar Linhas</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href=""></a><span class="fu">ggplot</span>(geneCounts, <span class="fu">aes</span>(<span class="at">x =</span> dex, <span class="at">y =</span> count, <span class="at">color =</span> cell, <span class="at">group =</span> cell)) <span class="sc">+</span></span>
<span id="cb81-2"><a href=""></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span> </span>
<span id="cb81-3"><a href=""></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb81-4"><a href=""></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb81-5"><a href=""></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#800000FF"</span>, <span class="st">"#767676FF"</span>, <span class="st">"#155F83FF"</span>, <span class="st">"#8A9045FF"</span>)) <span class="sc">+</span></span>
<span id="cb81-6"><a href=""></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="aula_5_rnaseq_24_files/figure-revealjs/graf_linha-1.png" style="width:100.0%" class="r-stretch"></section>
<section id="gráfico-ma" class="slide level2">
<h2>Gráfico MA</h2>
<ul>
<li><p>MA - “<em>mean average</em>” – log2 fold changes (no eixo y) x a média das contagens normalizada</p></li>
<li><p>Mostra quanto variância tem no modelo</p></li>
<li><p>Cores padronizados:</p>
<ul>
<li><p>Azul - genes com fold change significativa</p></li>
<li><p>Cinza - genes com fold change não - significativa</p></li>
</ul></li>
<li><p>Também chamado gráfico Bland-Altman</p></li>
</ul>
</section>
<section id="passo-anterior" class="slide level2 smaller">
<h2>Passo Anterior</h2>
<ul>
<li><p>Precisa restritar (“<em>shrink</em>”) os valores dos log2 fold changes para comparison das amostras tratadas contra as amostras não tratadas</p></li>
<li><p>Função <code>lfcShrink()</code> usa um dos 3 algoritmos para fazer esta restrição</p></li>
<li><p>Nós vamos usar o algoritmo <code>apeglm</code>, que fica no pacote <code>apeglm</code></p></li>
<li><p>Aplicar este algoritmo para a comparação que nos interesse – <code>dex_trt_vs_untrt</code></p></li>
<li><p>Achamos este nome no resultado da função resultsNames()</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href=""></a><span class="fu">library</span>(apeglm)</span>
<span id="cb82-2"><a href=""></a><span class="fu">resultsNames</span>(dds_out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"               "cell_N061011_vs_N052611"
[3] "cell_N080611_vs_N052611" "cell_N61311_vs_N052611" 
[5] "dex_trt_vs_untrt"       </code></pre>
</div>
</div>
</section>
<section id="lfcshrink-e-gráfico" class="slide level2 smaller">
<h2>lfcShrink e Gráfico</h2>
<ul>
<li><p>Agoritmo de shrink utiliza uma procedimento Bayesiano para suavizar contagens muito baixas ou com uma variância alta.</p></li>
<li><p>Gráfico mostra que como os normalized counts aumentam os genes com fold change insignificativa diminuem</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href=""></a>res_shrink <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_out, <span class="at">coef=</span><span class="st">"dex_trt_vs_untrt"</span>, <span class="at">type=</span><span class="st">"apeglm"</span>)</span>
<span id="cb84-2"><a href=""></a><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="aula_5_rnaseq_24_files/figure-revealjs/shrink2-1.png" style="width:100.0%" class="r-stretch"></section></section>
<section>
<section id="o-que-fazer-com-esses-resultados" class="title-slide slide level1 center">
<h1>O Que Fazer com Esses Resultados</h1>

</section>
<section id="relatórios-e-anotacões" class="slide level2">
<h2>Relatórios e Anotacões</h2>
<ul>
<li><p>Até agora, nossos resultados são muitos abstratos</p></li>
<li><p>Não tem como descrever eles em termos biológicos ainda</p></li>
<li><p>O processo de anotação torna esses tabelas dos genes ser mais compreensíveis</p></li>
</ul>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Note</strong></p>
</div>
<div class="callout-content">
<p>Os nomes dos genes que temos agora são ids dos genes <code>Ensembl</code> : e.g.&nbsp;ENSG00000285953.1</p>
</div>
</div>
</div>
</section>
<section id="anotacação-em-bioconductor" class="slide level2 smaller">
<h2>Anotacação em Bioconductor</h2>
<ul>
<li><p>Pacote <code>AnnotationDbi</code> – relaciona os nomes que já temos para nomes e atributos em uso comum</p></li>
<li><p>Nosso tecido vem de seres humanos</p></li>
<li><p>O pacote que lista os atributos dos genes humanos: <code>org.Hs.eg.db</code></p>
<ul>
<li>organismo/Homo Sapiens/Entrez Gene IDs/base de dados</li>
</ul></li>
<li><p>Mas, esses têm bastante informação</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href=""></a><span class="fu">library</span>(<span class="st">"AnnotationDbi"</span>)</span>
<span id="cb85-2"><a href=""></a><span class="fu">library</span>(<span class="st">"org.Hs.eg.db"</span>)</span>
<span id="cb85-3"><a href=""></a><span class="fu">columns</span>(org.Hs.eg.db)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"
 [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    
[11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         
[16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        
[21] "PMID"         "PROSITE"      "REFSEQ"       "SYMBOL"       "UCSCKG"      
[26] "UNIPROT"     </code></pre>
</div>
</div>
</section>
<section id="chave-para-criar-novas-colunas-de-dados-mapids" class="slide level2">
<h2>Chave para Criar Novas Colunas de Dados: <code>mapIds()</code></h2>
<ul>
<li><p><code>mapIDs()</code> entre na base de dados da espécie (<code>org.Hs.eg.db</code>) e deixa você tira as informações que quiser</p></li>
<li><p>Cada chamada para a função pode criar uma nova coluna em nossos resultados</p></li>
<li><p>Porque os IDs de nossos genes são do tipo ENSEMBL, esse deve ser nosso <code>keytype = "ENSEMBL"</code></p></li>
</ul>
</section>
<section id="criar-2-novas-colunas" class="slide level2">
<h2>Criar 2 novas colunas</h2>
<ol type="1">
<li><p><code>symbol</code> – ID que é mais comum invés de ENSEMBL</p></li>
<li><p><code>entrez</code> – ID do sistema ENTREZ</p></li>
</ol>
<div class="callout callout-note callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>ENTREZ</strong></p>
</div>
<div class="callout-content">
<p>Sistema de NCBI dos NIH que alberga informações sobre os nucleotídeos, proteinas, genes, e mais</p>
</div>
</div>
</div>
</section>
<section id="programação-de-mapids" class="slide level2 smaller">
<h2>Programação de <code>mapIDs()</code></h2>
<ul>
<li><p>Limitar para primeiros 15 genes em nossos resultados</p></li>
<li><p>Precisa todos os argumentos</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href=""></a><span class="co"># 1o 15 resultados</span></span>
<span id="cb87-2"><a href=""></a>first_fifteen <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(res), <span class="dv">1</span>, <span class="dv">15</span>)</span>
<span id="cb87-3"><a href=""></a><span class="co"># symbol</span></span>
<span id="cb87-4"><a href=""></a>res<span class="sc">$</span>symbol <span class="ot">&lt;-</span>  <span class="fu">mapIds</span>(org.Hs.eg.db,</span>
<span id="cb87-5"><a href=""></a>                      <span class="at">keys =</span> first_fifteen,</span>
<span id="cb87-6"><a href=""></a>                      <span class="at">column =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb87-7"><a href=""></a>                      <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb87-8"><a href=""></a>                      <span class="at">multiVals =</span> <span class="st">"first"</span>) <span class="co">#se tiver valores múltiplas</span></span>
<span id="cb87-9"><a href=""></a>res<span class="sc">$</span>entrez <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db,</span>
<span id="cb87-10"><a href=""></a>              <span class="at">keys =</span> first_fifteen,</span>
<span id="cb87-11"><a href=""></a>              <span class="at">column =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb87-12"><a href=""></a>              <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb87-13"><a href=""></a>              <span class="at">multiVals =</span> <span class="st">"first"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ver-os-novos-dados-dos-resultados-em-str" class="slide level2">
<h2>Ver os Novos Dados dos Resultados – em <code>str()</code></h2>

<img data-src="../images/symbolentrez.png" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="novos-resultados" class="slide level2 smaller">
<h2>Novos Resultados</h2>
<ul>
<li>Foco nos 6 mais importantes genes</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href=""></a>resOrdered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj), <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">6</span><span class="sc">:</span><span class="dv">8</span>)]</span>
<span id="cb88-2"><a href=""></a><span class="fu">head</span>(resOrdered, <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): dex trt vs untrt 
 
DataFrame with 6 rows and 5 columns
                    baseMean log2FoldChange         padj      symbol
                   &lt;numeric&gt;      &lt;numeric&gt;    &lt;numeric&gt; &lt;character&gt;
ENSG00000189221.9   2371.265        3.39426 3.95806e-134        MAOA
ENSG00000120129.5   3417.255        2.96990 2.75226e-129       DUSP1
ENSG00000101347.9  14106.720        3.74934 8.89591e-124      SAMHD1
ENSG00000152583.12   973.479        4.50022 9.04071e-112     SPARCL1
ENSG00000196136.17  2708.309        3.24329 3.94293e-109    SERPINA3
ENSG00000211445.11 12502.886        3.76804 6.73085e-108        GPX3
                        entrez
                   &lt;character&gt;
ENSG00000189221.9         4128
ENSG00000120129.5         1843
ENSG00000101347.9        25939
ENSG00000152583.12        8404
ENSG00000196136.17          12
ENSG00000211445.11        2878</code></pre>
</div>
</div>
</section></section>
<section id="só-uma-introdução-a-rnaseq-e-expressão-genômica" class="title-slide slide level1 center" data-background-color="blue">
<h1>Só Uma Introdução a RNASeq e Expressão Genômica</h1>

</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="aula_5_rnaseq_24_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>